File: python/snippets/recipes/enabled_not_suspended_last_login.py
"""Filter accounts that are enabled, not suspended, and have a last_login set.

FlyQL query: ``enabled and not suspended and last_login``

All three terms use truthy/falsy semantics: a bare key (no operator, no value)
means "this field is truthy". ``not suspended`` flips the truthy check into a
falsy check.
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse("enabled and not suspended and last_login")

columns = {
    "enabled": Column("enabled", "Bool"),
    "suspended": Column("suspended", "Bool"),
    "last_login": Column("last_login", "DateTime"),
}

sql = to_sql_where(result.root, columns)
print(sql)

File: javascript/packages/flyql/snippets/recipes/enabled-not-suspended-last-login.mjs
// Filter accounts that are enabled, not suspended, and have a last_login set.
//
// FlyQL query: enabled and not suspended and last_login
//
// All three terms use truthy/falsy semantics: a bare key (no operator, no
// value) means "this field is truthy". `not suspended` flips the truthy check
// into a falsy check.

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse('enabled and not suspended and last_login')

const columns = {
    enabled: newColumn({ name: 'enabled', type: 'Bool' }),
    suspended: newColumn({ name: 'suspended', type: 'Bool' }),
    last_login: newColumn({ name: 'last_login', type: 'DateTime' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)
